src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -124,10 +124,7 @@ public RecoverableSyntaxRoot<TRoot> WithSyntaxTree(IRecoverableSyntaxTree<TRoot>
 
                 using (var stream = await _storage.ReadStreamAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    var retVal = RecoverRoot(stream, cancellationToken);
-                    _storage.Dispose();
-                    _storage = null;
-                    return retVal;
+                    return RecoverRoot(stream, cancellationToken);
```

Comment:
If _storage cannot be disposed here, what's the right place to call _storage.Dispose()?